libseccomp/tests/tests.rs

```diff
@@ -0,0 +1,317 @@
+    let mut ctx = ScmpFilterContext::new_filter(ScmpAction::KillThread).unwrap();
+    let syscall = ScmpSyscall::from_name("open").unwrap();
+    let priority = 100;
+
+    assert!(ctx.set_syscall_priority(syscall, priority).is_ok());
+    assert!(ctx.set_syscall_priority(-1, priority).is_err());
+}
+
+#[test]
+fn test_filter_attributes() {
+    let mut ctx = ScmpFilterContext::new_filter(ScmpAction::KillThread).unwrap();
+
+    // Test for CtlNnp
+    ctx.set_ctl_nnp(false).unwrap();
+    let ret = ctx.get_ctl_nnp().unwrap();
+    assert!(!ret);
+
+    // Test for ActBadArch
+    let test_actions = [
+        ScmpAction::Trap,
+        ScmpAction::Errno(libc::EACCES),
+        ScmpAction::Trace(10),
+    ];
+    for action in test_actions {
+        ctx.set_act_badarch(action).unwrap();
+        let ret = ctx.get_act_badarch().unwrap();
+        assert_eq!(ret, action);
+    }
+
+    // Test for ActDefault
+    let ret = ctx.get_act_default().unwrap();
+    assert_eq!(ret, ScmpAction::KillThread);
+
+    // Test for CtlLog
+    if check_api(3, ScmpVersion::from((2, 4, 0))).unwrap() {
+        ctx.set_ctl_log(true).unwrap();
+        let ret = ctx.get_ctl_log().unwrap();
+        assert!(ret);
+    } else {
+        assert!(ctx.set_ctl_log(true).is_err());
+        assert!(ctx.get_ctl_log().is_err());
+    }
+
+    // Test for CtlSsb
+    if check_api(4, ScmpVersion::from((2, 5, 0))).unwrap() {
+        ctx.set_ctl_ssb(true).unwrap();
+        let ret = ctx.get_ctl_ssb().unwrap();
+        assert!(ret);
+    } else {
+        assert!(ctx.set_ctl_ssb(true).is_err());
+        assert!(ctx.get_ctl_ssb().is_err());
+    }
+
+    // Test for CtlOptimize
+    let opt_level = 2;
+    if check_api(4, ScmpVersion::from((2, 5, 0))).unwrap() {
+        ctx.set_ctl_optimize(opt_level).unwrap();
+        let ret = ctx.get_ctl_optimize().unwrap();
+        assert_eq!(ret, opt_level);
+    } else {
+        assert!(ctx.set_ctl_optimize(opt_level).is_err());
+        assert!(ctx.get_ctl_optimize().is_err());
+    }
+
+    // Test for ApiSysRawRc
+    if check_api(4, ScmpVersion::from((2, 5, 0))).unwrap() {
+        ctx.set_api_sysrawrc(true).unwrap();
+        let ret = ctx.get_api_sysrawrc().unwrap();
+        assert!(ret);
+    } else {
+        assert!(ctx.set_api_sysrawrc(true).is_err());
+        assert!(ctx.get_api_sysrawrc().is_err());
+    }
+
+    // Test for CtlTsync
+    if check_api(2, ScmpVersion::from((2, 2, 0))).unwrap() {
+        ctx.set_ctl_tsync(true).unwrap();
+        let ret = ctx.get_ctl_tsync().unwrap();
+        assert!(ret);
+    } else {
+        assert!(ctx.set_ctl_tsync(true).is_err());
+        assert!(ctx.get_ctl_tsync().is_err());
+    }
+}
+
+#[test]
+fn test_filter_reset() {
+    let mut ctx = ScmpFilterContext::new_filter(ScmpAction::KillThread).unwrap();
+    ctx.reset(ScmpAction::Allow).unwrap();
+
+    let action = ctx.get_act_default().unwrap();
+
+    let expected_action = ScmpAction::Allow;
```

Comment:
```diff
-        let action = ctx.get_filter_attr(ScmpFilterAttr::ActDefault).unwrap();
+        let action = ctx.get_act_default().unwrap();
 
-        let expected_action: u32 = ScmpAction::Allow.to_sys();
+        let expected_action = ScmpAction::Allow;
```

TODO: does this affect coverage?

Reply:
It doesn't affect coverage because `get_act_default` calls `get_filter_attr(ScmpFilterAttr::ActDefault)`.

Reply:
Actually I was thinking about `ScmpAction::to_sys`.

Reply:
Which is done in `test_parse_action()`.